lib/portlayer/storage/vsphere/volume.go

```diff
@@ -191,3 +193,57 @@ func (v *VolumeStore) VolumesList(op trace.Operation) ([]*storage.Volume, error)
 
 	return volumes, nil
 }
+
+// ReadArchive reads the delta between child and parent volume layers, returning
+// the difference as a tar archive.
+//
+// store - the volume store containing the two layers
+// child - the leaf layer of the volume chain
+// ancestor - the volume layer up the chain against which to diff
+// pathspec - describes filters on paths found in the data (include, exclude, strip)
+// data - set to true to include file data in the tar archive, false to include headers only
+func (v *VolumeStore) ReadArchive(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
+	_, err := util.VolumeStoreName(store)
+	if err != nil {
+		return nil, err
+	}
+
+	if ancestor == "" {
+		ancestor = storage.Scratch.ID
+	}
+
+	c := v.volDiskDSPath(child)
+	a := v.volDiskDSPath(ancestor)
+
+	cleanFunc := func() {
+		if err := v.dm.UnmountAndDetach(op, c); err != nil {
+			op.Infof("Error cleaning up child disk: %s", err.Error())
+		}
+		if err = v.dm.UnmountAndDetach(op, a); err != nil {
+			op.Infof("Error cleaning up ancestor disk: %s", err.Error())
+		}
+	}
+
+	childFs, err := v.dm.AttachAndMount(op, c)
+	if err != nil {
+		return nil, err
+	}
+
+	ancestorFs, err := v.dm.AttachAndMount(op, a)
```

Comment:
@caglar10ur we need to look into how we can combine these operations - there's no reason to have multiple vsphere calls for what should be a single reconfigure.